src/glib/base/pgblob.cpp

```diff
@@ -592,13 +616,22 @@ TPgBlobPt TPgBlob::Put(
 		memcpy(PgBf + ItemRec->Offset, Bf, BfL);
 		return Pt;
 
-	} else if (existing_size + PgH->GetFreeMem() >= BfL) {
+	} else if (existing_size + PgH->GetFreeMem() >= BfL + (int)sizeof(TPgBlobPageItem)) {
 		// ok, everything can still be inside this page
 		DeleteItem(PgBf, Pt.GetIIx());
+        //Fsm.FsmUpdatePage(PgPt, PgH->GetFreeMem());
+
 		ChangeItem(PgBf, Pt.GetIIx(), Bf, BfL);
 		Fsm.FsmUpdatePage(Pt, PgH->GetFreeMem());
 		return Pt;
 
+        //uint16 ii = AddItem(PgBf, Bf, BfL);
```

Comment:
Same here, unless there is some good reason for keeping this code, if yes it should be noted why in the comments.